Tests/GeneratorTests.cs

```diff
@@ -101,5 +100,23 @@ public void TestGenerateWikiEventsList()
             Directory.CreateDirectory(@"Wiki\");
             File.WriteAllLines(@"Wiki\Events.md", output);
         }
+
+        [TestMethod, TestCategory("DocGen")]
+        // Generates the list of keywords used by the Cottle grammar "Cottle.xshd".
+        public void TestGenerateEventVariables()
+        {
+            SortedSet<string> eventVars = new SortedSet<string>();
+            foreach (IDictionary<string, string> variableList in Events.VARIABLES.Values)
+            {
+                foreach (string variableName in variableList.Keys)
+                {
+                    eventVars.Add(variableName);
+                }
+            }
+
+            string output = "      <Word>" + string.Join("</Word>\r\n      <Word>", eventVars) + " </Word>\r\n";
+            Directory.CreateDirectory(@"Cottle\");
+            File.WriteAllText(@"Cottle\Custom keywords.txt", output);
+        }
```

Comment:
Should this file include a header with instructions? Something like "Paste the generated text below into the "Custom properties" rule set in `SpeechResponder\Cottle.xshd`"?

Reply:
It's probably not very possible without revamping how we define custom functions in Cottle, but I assume you also considered generating a list of custom functions?

Reply:
I think I'll put instructions in a comment rather than in the output.

I did look into generating the list of custom functions but it would have involved more restructuring than I was willing to take on for this iteration. So I just extracted the names manually using Notepad++ and some creative use of regex. I still hope to do it though.

Reply:
One major change at a time? Fair enough. 😉